junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -28,6 +29,8 @@
     private final CountDownLatch startable;
     private final CountDownLatch finished;
     private volatile Thread testThread;
+    
+    private final AtomicBoolean exceptionOccurred = new AtomicBoolean(false); 
```

Comment:
Regarding the changes you suggest in await/notify barrier, I don't agree. If the connection is lost and we ignore the exception, then the flow in the test will continue and we will get an exception in another place. The new exception might be misleading, not reflecting the actual situation encountered. Also, in awaitBarrier, if we ignore the error we might get stuck in waiting for the latch that will never be triggered.

I don't think we should ignore exceptions, but I guess we need to change the code to properly handle them - stop communication thread and throw exception. Calling latch.notifyException might not be the best way to handle the communication error. 

I think we should open another issue for this change, to explore in more depth the changes required.